src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala

```diff
@@ -576,5 +592,154 @@ class AdminSpec extends BaseSpec with OptionValues with CancelAfterFailure with
         }
       }
     }
+
+    "listing projects" should {
+
+      "delete all ACLs for user" in cleanAcls
+
+      "return unauthorized access if user has no permissions on / or particular project" in {
+        eventually {
+          cl(Req(GET, s"$adminBase/projects?size=1", headersUser, Json.obj().toEntity)).mapJson { (json, result) =>
+            result.status shouldEqual StatusCodes.Unauthorized
+            json shouldEqual jsonContentOf("/iam/errors/unauthorized-access.json", errorCtx)
+          }
+        }
+      }
+
+      "add projects/create permissions for user" in {
+        val json = jsonContentOf("/iam/add.json", replSub + (quote("{perms}") -> "projects/create")).toEntity
+        cl(Req(PUT, s"$iamBase/acls/", headersGroup, json)).mapResp { result =>
+          result.status shouldEqual StatusCodes.OK
+          result.entity.isKnownEmpty() shouldEqual true
+        }
+
+        eventually {
+          cl(Req(uri = s"$iamBase/acls/", headers = headersUser)).mapJson { (json, result) =>
+            result.status shouldEqual StatusCodes.OK
+            json shouldEqual jsonContentOf(
+              "/iam/project-perms-response.json",
+              resourceIamCtx ++ replSub + (quote("{perms}") -> "projects/create", quote("{path}") -> "")
+            )
+          }
+        }
+      }
+
+      val projectIds: immutable.Seq[String] = 1 to 3 map { _ =>
+        genId()
+      }
+
+      def projectListingResults(ids: Seq[String]): Json = {
+        Json.arr(
+          ids.sorted.map { id =>
+            Json.obj(
+              "_id" -> Json.fromString(s"$adminBase/projects/$id"),
+              "_source" -> Json.obj(
+                "@id" -> Json.fromString(s"$adminBase/projects/$id")
+              )
+            )
+          }: _*
+        )
+      }
+
+      def projectListingResultsWithAllFields(ids: Seq[String]): Json = {
```

Comment:
Can't you use a json template (json file with replacements) as we did in other places. In my opinion that increases readability

Reply:
will do